File: bindings/lua/Polycode/SoundManager.lua
class "SoundManager"


function SoundManager:SoundManager(...)
	local arg = {...}
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.SoundManager(unpack(arg))
	end
end

function SoundManager:setListenerPosition(position)
	local retVal = Polycode.SoundManager_setListenerPosition(self.__ptr, position.__ptr)
end

function SoundManager:setListenerOrientation(orientation)
	local retVal = Polycode.SoundManager_setListenerOrientation(self.__ptr, orientation.__ptr)
end

function SoundManager:recordSound(rate, sampleSize)
	local retVal = Polycode.SoundManager_recordSound(self.__ptr, rate, sampleSize)
	return retVal
end

function SoundManager:Update()
	local retVal =  Polycode.SoundManager_Update(self.__ptr)
end

function SoundManager:setGlobalVolume(globalVolume)
	local retVal = Polycode.SoundManager_setGlobalVolume(self.__ptr, globalVolume)
end

function SoundManager:__delete()
	if self then Polycode.delete_SoundManager(self.__ptr) end
end

File: bindings/javascript/Polycode/SoundManager.js
function SoundManager() {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.SoundManager()
	}
}


Duktape.fin(SoundManager.prototype, function (x) {
	if (x === SoundManager.prototype) {
		return;
	}
	Polycode.SoundManager__delete(x.__ptr)
})

SoundManager.prototype.setListenerPosition = function(position) {
	Polycode.SoundManager_setListenerPosition(this.__ptr, position)
}

SoundManager.prototype.setListenerOrientation = function(orientation) {
	Polycode.SoundManager_setListenerOrientation(this.__ptr, orientation)
}

SoundManager.prototype.recordSound = function(rate,sampleSize) {
	return Polycode.SoundManager_recordSound(this.__ptr, rate, sampleSize)
}

SoundManager.prototype.Update = function() {
	Polycode.SoundManager_Update(this.__ptr)
}

SoundManager.prototype.setGlobalVolume = function(globalVolume) {
	Polycode.SoundManager_setGlobalVolume(this.__ptr, globalVolume)
}
